### DISCUSSION TITLE
How to implement service - repository pattern?

### DISCUSSION THREAD

**Discussion by iamshulz** [adonisjs/core#3269] [Help]
How to implement either service or repository (or both) on AdonisJs in order to resuse and prevent bloating of controller?

**Comment by RomainLanz** [comment]
Hey @jamols09! 👋🏻 

What have you tried so far?

**Reply by iamshulz** [reply]
Unfortunately I have not yet, coming from laravel and Im trying to learn a the world of js I noticed that are are a lot of backend framework using js. I'm still trying to choose what framework fits me but this one is promising and it's very familiar but I have not seen a service/repository pattern implementation yet (although laravel also is on the docs).

But it would be great if you have any sources
